### ISSUE TITLE
When setting Python version to 3.10 using runtime.txt, you actually get v3.7

### ISSUE THREAD

**Issue by sgibson91** [jupyterhub/repo2docker#1183]
### Bug description

_Note: This bug was originally surfaced via a help request in gitter._

When you use a `runtime.txt` file to set the Python version, and request v3.10, you actually get version 3.7 (I suspect the default version r2d is configured to install?)

#### Expected behaviour

The Python version stated in the `runtime.txt` file is respected and Python version 3.10 is installed.

#### Actual behaviour

Instead of 3.10, Python 3.7 is available.

### How to reproduce

I created an MWE repo to demonstrate this: https://github.com/sgibson91/pythonv3.10-binder-mwe

1. Launch the above repo in binder: [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/sgibson91/pythonv3.10-binder-mwe/HEAD)
2. Open a Terminal
3. Execute the command `python --version`
4. See result is `3.7.*`, not `3.10.*`

Alternatively:

2. Open a new Notebook
3. Execute the below code in a cell:

```
import sys
print(sys.version)
```

4. See result is `3.7.*`, not `3.10.*`

### Your personal set up

See files in https://github.com/sgibson91/pythonv3.10-binder-mwe for setup details.

**Comment by betatim** [comment]
I thought the issue here is that there is no `environment-py3.10.yml` in the conda buildpack, but there is. Has someone tried building this repo locally with repo2docker `main`? The thing to look out for is a message right at the start of the build process which tells you which buildpack (and Python verison) was selected. This helps figuring out if the wrong buildpack is selected or if the correct buildpack is used but for some reason the wrong Python version is installed/accessible.

**Comment by manics** [comment]
Python 3.10 is a very recent addition to r2d. mybinder.org deploys are currently blocked
https://github.com/jupyterhub/mybinder.org-deploy/issues/2337
so it may not be deployed.

If this is the problem then it would be nice if r2d failed with an unsupported python version instead of silently falling back to the default.

**Comment by minrk** [comment]
We should also fail informatively on an unrecognized/unsuppported Python, not fallback on default

**Comment by sgibson91** [comment]
1. I installed r2d from `main` using `python3 -m pip install https://github.com/jupyterhub/repo2docker/archive/main.zip` (into a fresh conda env)
2. I locally cloned https://github.com/sgibson91/pythonv3.10-binder-mwe
3. I ran `repo2docker .` from the repo root
4. Logs from r2d:

```
Picked Local content provider.
Using local repo ..
Building conda environment for python=3.10
Using PythonBuildPack builder
Building conda environment for python=3.10
Building conda environment for python=3.10
```

This time when I ran `python --version`, I got version 3.10. So this is a binder issue, not a r2d issue. Apologies for the noise! Shall I transfer this to the mybinder.org-deploy repo?

**Comment by minrk** [comment]
As @manics mentioned, this was fixed very recently in repo2docker (#1175), but that update had yet to be deployed to mybinder.org due to https://github.com/jupyterhub/mybinder.org-deploy/issues/2337. That was just fixed and repo2docker updated by merging https://github.com/jupyterhub/mybinder.org-deploy/pull/2336, so the issue for 3.10 in particular should be resolved for both repo2docker and mybinder.org. But image caching means you'll need to trigger a rebuild by pushing a new commit. This should work, I think: https://gke.mybinder.org/v2/gh/minrk/pythonv3.10-binder-mwe/HEAD

I think we can close this, as #552 already represents the issue of incorrect envs instead of errors for unsupported Python (fixed in #1184)

**Comment by sgibson91** [comment]
Thanks! Happy to close
